packages/web-components/src/menu-item/menu-item.ts

```diff
@@ -183,23 +130,20 @@ export class MenuItem extends FASTElement {
 
       case keyArrowRight:
         //open/focus on submenu
-        this.expanded && this.submenu ? this.submenu.focus() : this.expandAndFocus();
+        if (this.disabled) return false;
+        this.submenu?.togglePopover(true);
+        this.submenu?.focus();
         return false;
 
-      case keyEscape:
-        // close submenu
-        if (this.expanded) {
-          this.closeSubMenu();
-          return false;
-        }
-        break;
-
       case keyArrowLeft:
         //close submenu
-        if (this.expanded) {
-          this.closeSubMenu();
-          return false;
+        try {
+          this.parentElement?.togglePopover(false);
+        } catch (e) {
+          // Catch DOMException if parentElement is not a menu
```

Comment:
Maybe check if `parentElement` is a menu instead of using try...catch? I think it'd make the code more readable and explicit.